riscv-c-api.md

```diff
@@ -503,6 +504,22 @@ Sign extension of 32-bit values on RV64 is not reflected in the interface.
 | `uint32_t __riscv_sm4ed(uint32_t rs1, uint32_t rs2, const int bs);`     | `sm4ed`       | Zksed             | `bs`=[0..3] |
 | `uint32_t __riscv_sm4ks(uint32_t rs1, uint32_t rs2, const int bs);`     | `sm4ks`       | Zksed             | `bs`=[0..3] |
 
+### May-Be-Operations Extension Intrinsics
+
+The functions are only available if the compiler's `-march` string
+enables the required ISA extension. (Calling functions for not enabled
+ISA extensions will lead to compile-time and/or link-time errors.)
+
+Unsigned types are used as that is the most logical representation for a
+collection of bits.
+
+Sign extension of 32-bit values on RV64 is not reflected in the interface.
+
+| Prototype                                                               | Instruction   | Extension  | Notes          |
+|-------------------------------------------------------------------------|---------------|------------|----------------|
+| `uint32_t __riscv_mopr(uint32_t rs1, const int n);`                     | `mop.r.[n]`   | Zimop      | `n`=[0..31]    |
+| `uint32_t __riscv_moprr(uint32_t rs1, uint32_t rs2, const int n);`      | `mop.rr.[n]`  | Zimop      | `n`=[0..8]     |
```

Comment:
The interface seems not right to me, it may not work correctly if the address larger than 4G, it will truncated due to the type, so I guess `uintptr_t` may be better type, but maybe uintxlen_t is what we really need here since ... ilp32 for RV64.

cc. @topperc @cmuellner

Reply:
Or we follow the pattern that @topperc defined for the bitmanip and scalar crypto extensions: use `_32` and `_64` postfixes and restrict the availability of the `_64` API to RV64. Plus the following paragraph:
> Intrinsics operating on XLEN sized value are not available as there is no type defined. If xlen_t is added in the future, this can be revisited.

Reply:
I noticed that definitions of intrinsic on GCC and LLVM are differ.
For example intrinsic for clmul defined as __riscv_clmul in GCC and __riscv_clmul_32/__riscv_clmul_64 in LLVM.
Should I define with postfixes and implement in each compiler in own way or should implement with the same API?

Reply:
> Or we follow the pattern that @topperc defined for the bitmanip and scalar crypto extensions: use `_32` and `_64` postfixes and restrict the availability of the `_64` API to RV64. Plus the following paragraph:

@ln8-8 updated the patch as you requested, besides he updated patch for LLVM. 
Now it meets these requirements.